src/PrestaShopBundle/Translation/Provider/ThemeProvider.php

```diff
@@ -70,11 +90,33 @@ public function getMessageCatalogue()
     }
 
     /**
+     * @param null $baseDir
      * @return string Path to app/themes/{themeName}/translations/{locale}
      */
-    public function getResourceDirectory()
+    public function getResourceDirectory($baseDir = null)
     {
-        return $this->resourceDirectory.'/'.$this->themeName.'/translations/'.$this->getLocale();
+        if (is_null($baseDir)) {
+            $baseDir = $this->resourceDirectory;
+        }
+
+        $resourceDirectory = $baseDir.'/'.$this->themeName.'/translations/'.$this->getLocale();
+
+        if (!$this->filesystem->exists($resourceDirectory)) {
```

Comment:
You don't need to do the check http://symfony.com/doc/current/components/filesystem.html#mkdir :)